docs/spec_draft.md

```diff
@@ -77,6 +77,11 @@ are aligned, whether they are stored contiguously, etc).
 **Token type** Values of this type are used for imposing order on execution of
 side-effecting operations using data dependencies.
 
+**Tuple types** are ordered sets of elements of (possibly)different types.
+Referred in document as `tuple<T0, T2, ... TN-1>`, where `N` represents the size
+of the tuple and `Ti` are represent the type of the elements in the tuple, `Ti`
+can be of `token` type, `tuple` type, or a `tensor` of any supported type.
```

Comment:
How about alphabetically ordering? Also, I don't think "any supported type" is just restricted to tensors:
```
can be of `tensor`, `token`, or `tuple` of any supported type.
```